f2/src/utils/nextWhatWasAffectedUrl.js

```diff
@@ -1,16 +1,114 @@
-import { whatWasAffectedPages } from '../forms/WhatWasAffectedForm'
-
-export const nextWhatWasAffectedUrl = (optionsSelected, currentPage) => {
-  let pages = whatWasAffectedPages.filter(
-    page => page.url !== '' && optionsSelected.indexOf(page.key) > -1,
-  )
-  pages.unshift({ key: '', url: 'whatwasaffected' }) // start of list
-  pages.push({ key: '', url: 'whathappened' }) // end of list
-
-  for (const [index, page] of pages.entries()) {
-    if (page.url === currentPage) {
-      return pages[Math.min(index + 1, pages.length)].url
+const { getLogger } = require('./winstonLogger')
+
+const logger = getLogger(__filename)
+
+export const whatWasAffectedPages = {
+  FINANCIAL: {
+    key: 'whatWasAffectedForm.financial',
+    url: 'moneylost',
+  },
+  INFORMATION: {
+    key: 'whatWasAffectedForm.personalInformation',
+    url: 'information',
+  },
+  DEVICES: {
+    key: 'whatWasAffectedForm.devices',
+    url: 'devices',
+  },
+  BUSINESS: {
+    key: 'whatWasAffectedForm.business_assets',
+    url: 'business',
+  },
+  OTHER: {
+    key: 'whatWasAffectedForm.other',
+    url: '',
+  },
+  FIRST_PAGE: {
+    key: '',
+    url: 'whatwasaffected',
+  },
+  LAST_PAGE: {
+    key: '',
+    url: 'whathappened',
+  },
+  CONFIRMATION: {
+    key: '',
+    url: 'confirmation',
+  },
+}
+
+export const pages = {
+  firstPage: whatWasAffectedPages.FIRST_PAGE,
+  lastPage: whatWasAffectedPages.LAST_PAGE,
```

Comment:
The lastPage could be removed, it only holds two states, either Confirmation page or what happened page. It's initialized to be what happened page, but set to confirmation page once doneForms being true.
This would simply pages object, also you could remove 
  if (doneForms) {
    pages.lastPage = whatWasAffectedPages.CONFIRMATION
  }
from all affected pages.
And move the logic into nextWhatWasAffectedUrl
export const nextPage = (doneForms, navObject) => {
  try {
    const selectedOptions = navObject.affectedOptions
    const lastSelection = selectedOptions.slice(-1)[0]
    let nextPage
    let index

    //If there are no more pages to display proceed to What Happened or Confirmation.
    if (
      navObject.currentPage.key === lastSelection ||
      selectedOptions.length === 0
    ) {
      if (doneForms)  return whatWasAffectedPages.CONFIRMATION
      return whatWasAffectedPages.LAST_PAGE
    }

Reply:
This has been implemented.